extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/MicrometerTimedInterceptor.java

```diff
@@ -82,11 +71,40 @@ private Object processWithTimer(InvocationContext context, Timed timed, Tags com
             exceptionClass = MicrometerRecorder.getExceptionTag(ex);
             throw ex;
         } finally {
-            record(timed, metricName, sample, exceptionClass, timerTags);
+            for (Sample sample : samples) {
+                sample.stop(exceptionClass);
+            }
         }
     }
 
-    private void record(Timed timed, String metricName, Timer.Sample sample, String exceptionClass, Tags timerTags) {
+    private List<Sample> getSamples(InvocationContext context) {
+        Method method = context.getMethod();
+        Tags commonTags = getCommonTags(method.getDeclaringClass().getName(), method.getName());
+        List<Timed> timed = new ArrayList<>();
+        @SuppressWarnings("unchecked")
+        Set<Annotation> bindings = (Set<Annotation>) context.getContextData()
+                .get(ArcInvocationContext.KEY_INTERCEPTOR_BINDINGS);
+        for (Annotation annotation : bindings) {
+            if (annotation.annotationType().equals(Timed.class)) {
+                timed.add((Timed) annotation);
+            }
+        }
```

Comment:
again, pretty sure I have a common method for this lookup (for MP Metrics)... it may need to be moved